drivers/pds/lib/utils.go

```diff
@@ -1017,6 +1218,63 @@ func CreateRmqWorkload(dnsEndpoint string, pdsPassword string, namespace string,
 	return pod, nil
 }
 
+// This function prepares a deployment for running TPCC Workload
+func CreateTpccWorkloads(dataServiceName string, deploymentID string, scalefactor string, iterations string, deploymentName string, namespace string) (bool, error) {
+	var dbUser, timeToRun, numOfCustomers, numOfThreads, numOfWarehouses string
+
+	dnsEndpoint, err := GetDeploymentConnectionInfo(deploymentID)
+	if err != nil {
+		log.Errorf("An Error Occured while getting connection info %v", err)
+		return false, err
+	}
+	log.Infof("Dataservice DNS endpoint %s", dnsEndpoint)
+	pdsPassword, err := GetDeploymentCredentials(deploymentID)
+	if err != nil {
+		log.Errorf("An Error Occured while getting credentials info %v", err)
+		return false, err
+	}
+
+	switch dataServiceName {
+	// For a Postgres workload, simply create schema and run the TPCC Workload for default time
+	case postgresql:
+		dbName := "pds"
+		wasTpccRunSuccessful := RunTpccWorkload(dbUser, pdsPassword, dnsEndpoint, dbName,
+			timeToRun, numOfThreads, numOfCustomers, numOfWarehouses,
+			deploymentName, namespace, dataServiceName)
+		if !wasTpccRunSuccessful {
+			return wasTpccRunSuccessful, errors.New("Tpcc run failed. This could be a bug - please check manually")
+		} else {
+			return wasTpccRunSuccessful, nil
+		}
+	// For MySQL workload, first setup the deployment to run TPCC, then wait for MySQL to be available,
+	// Create TPCC Schema and then run it.
+	case mysql:
+		dbName := "tpcc"
+		var wasMysqlConfigured bool
+		// Waiting for approx an hour to check if Mysql deployment comes up
+		for i := 1; i <= 80; i++ {
+			wasMysqlConfigured := SetupMysqlDatabaseForTpcc(dbUser, pdsPassword, dnsEndpoint, namespace)
+			if wasMysqlConfigured {
+				log.InfoD("MySQL Deployment is successfully configured to run for TPCC Workload. Starting TPCC Workload Now.")
+				break
+			} else {
+				log.InfoD("MySQL deployment is not yet configured for TPCC. It may still be starting up or there could be some error")
+				log.InfoD("Waiting for some more time to see if it can be reached within 30 minutes")
+				time.Sleep(30 * time.Second)
+			}
+		}
+		if !wasMysqlConfigured {
+			log.Errorf("Something went wrong and DB Couldn't be prepared for TPCC workload. Exiting.")
+			return wasMysqlConfigured, errors.New("MySQL DB Couldnt be prepared for TPCC as it wasnt reachable. This could be a bug, please check manually.")
+		}
+		wasTpccRunSuccessful := RunTpccWorkload(dbUser, "password", dnsEndpoint, dbName,
+			timeToRun, numOfThreads, numOfCustomers, numOfWarehouses,
+			deploymentName, namespace, dataServiceName)
+		return wasTpccRunSuccessful, errors.New("TPCC Run failed. This could be a bug - please check manually")
```

Comment:
same as above

Reply:
Done - incorporated this in the Runtpccworkload method